### DISCUSSION TITLE
Should createClient be called only once or many times?

### DISCUSSION THREAD

**Discussion by kotx** [edgedb/edgedb-js#414] [Q&A] [answered]
Also if I were to have globals like a `current_user_id` as defined in the docs, should I call `withGlobals()` on the single client instance returned from `createClient` every time or should I call` createClient().withGlobals()` every time?
Using Nuxt.js 3 with the state accessed from a plugin, if that matters.

**Comment by elprans** [comment, marked as answer]
`createClient` should be called once per app.  This can be done globally.  `withGlobals()` returns a shallow wrapper over a client with configured globals and should be called whenever you want to set a global variable, i.e. in a request middleware or request handler.

**Reply by kotx** [reply]
Got it, thanks!
